Unexpected urllib.error.URLError message in test_download_ftp_file_properly_handles_socket_error

### Description
<!-- Provide a general description of the bug. -->
The expected exceptions in
https://github.com/astropy/astropy/blob/2bdbe28ee8b047a2043fd6a3582c587a00e298f7/astropy/utils/tests/test_data.py#L2168
do not include a message `<urlopen error [Errno 8] nodename nor servname provided, or not known.` raised under OS X 10.14 with Python 3.7-3.9.

### Actual behavior
<!-- What actually happened. -->
<!-- Was the output confusing or poorly described? -->
```
_______________________________________________________________ test_download_ftp_file_properly_handles_socket_error _______________________________________________________________

    @pytest.mark.remote_data
    def test_download_ftp_file_properly_handles_socket_error():
        faulty_url = "ftp://anonymous:mail%40astropy.org@nonexisting/pub/products/iers/finals2000A.all"
        with pytest.raises(urllib.error.URLError) as excinfo:
            download_file(faulty_url)
        errmsg = excinfo.exconly()
>       assert "Name or service not known" in errmsg or 'getaddrinfo failed' in errmsg or 'Temporary failure in name resolution' in errmsg  # noqa
E       AssertionError: assert ('Name or service not known' in 'urllib.error.URLError: <urlopen error [Errno 8] nodename nor servname provided, or not known. requested URL: ftp://anonymous:mail%40astropy.org@nonexisting/pub/products/iers/finals2000A.all>' or 'getaddrinfo failed' in 'urllib.error.URLError: <urlopen error [Errno 8] nodename nor servname provided, or not known. requested URL: ftp://anonymous:mail%40astropy.org@nonexisting/pub/products/iers/finals2000A.all>' or 'Temporary failure in name resolution' in 'urllib.error.URLError: <urlopen error [Errno 8] nodename nor servname provided, or not known. requested URL: ftp://anonymous:mail%40astropy.org@nonexisting/pub/products/iers/finals2000A.all>')

/opt/lib/python3.7/site-packages/astropy/utils/tests/test_data.py:2168: AssertionError
```
### Steps to Reproduce
<!-- Ideally a code example could be provided so we can run it ourselves. -->
<!-- If you are pasting code, use triple backticks (```) around
your code snippet. -->
<!-- If necessary, sanitize your screen output to be pasted so you do not
reveal secrets like tokens and passwords. -->

```python
import astropy
astropy.test(package='utils', args='-vk _ftp_file', remote_data=True)
```

### System Details
<!-- Even if you do not think this is necessary, it is useful information for the maintainers. -->
macOS-10.14.6-x86_64-i386-64bit

Python 3.7.9 (default, Oct 14 2020, 22:54:20) [Clang 11.0.0 (clang-1100.0.33.17)] on darwin
Python 3.8.6 (default, Oct 15 2020, 01:14:04) [Clang 11.0.0 (clang-1100.0.33.17)] on darwin
Python 3.9.0 (default, Oct 10 2020, 10:27:03) [Clang 11.0.0 (clang-1100.0.33.17)] on darwin

Numpy 1.18.5/1.19.4
astropy 4.2rc1
Scipy 1.5.3
Matplotlib 3.3.2

--- a/astropy/utils/tests/test_data.py
+++ b/astropy/utils/tests/test_data.py
@@ -2165,7 +2165,16 @@ def test_download_ftp_file_properly_handles_socket_error():
     with pytest.raises(urllib.error.URLError) as excinfo:
         download_file(faulty_url)
     errmsg = excinfo.exconly()
-    assert "Name or service not known" in errmsg or 'getaddrinfo failed' in errmsg or 'Temporary failure in name resolution' in errmsg  # noqa
+    found_msg = False
+    possible_msgs = ['Name or service not known',
+                     'nodename nor servname provided, or not known',
+                     'getaddrinfo failed',
+                     'Temporary failure in name resolution']
+    for cur_msg in possible_msgs:
+        if cur_msg in errmsg:
+            found_msg = True
+            break
+    assert found_msg, f'Got {errmsg}, expected one of these: {",".join(possible_msgs)}'
 
 
 @pytest.mark.parametrize(

